tests/functional/test-lease-expire.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+project_id=75681640118a4890b3da0d106eae8af7 # test1 uuid
+tmpfile=$(mktemp ./leaseXXXXXX)
+start=$(date +%Y-%m-%d)
+end=$(date -d "+5 days" +"%Y-%m-%d")
+
+# Setup script, contains offer create test
+
+./setup.sh "$tmpfile" "$project_id" "$start" "$end" "$resource_type"
+
+. $tmpfile
+
+# The uuid variable will be overwritten, so we save it here
+test_offer_uuid=$uuid
+# Run cleanup if the script exits, contains offer delete test
+trap "./cleanup.sh $test_offer_uuid; rm -f $tmpfile $errfile $nodefile" EXIT
+
+######################
+## OFFER CLAIM TEST ##
+######################
+
+# Tests that a lessee can claim an offer available to them
+
+echo "OFFER CLAIM TEST"
+
+openstack --os-cloud test1-subproject esi offer claim $uuid \
+  --start-time="$start" \
+  --end-time="$(date -d "+1 minute" +"%Y-%m-%d %H:%M:%S")" \
+  -f shell > $tmpfile \
+  || { ec=$?; echo "ERROR: failed to claim offer" >&2; exit $ec; }
+
+echo "OFFER CLAIM TEST SUCCEEDED"
+
+sleep 2m
```

Comment:
You may want to print out a statement that the test will pause for 2 minutes, just so someone running the test knows what to expect.

Reply:
Sounds good